pkg/disks/raid.go

```diff
@@ -0,0 +1,134 @@
+package disks
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/scylladb/scylla-operator/pkg/util/blkutils"
+	oexec "github.com/scylladb/scylla-operator/pkg/util/exec"
+	"github.com/scylladb/scylla-operator/pkg/util/slices"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/exec"
+)
+
+func MakeRAID0(ctx context.Context, executor exec.Interface, sysfsPath string, raidDevice string, devices []string, udevControlEnabled bool) error {
+	if _, err := os.Stat(raidDevice); !os.IsNotExist(err) {
+		klog.InfoS("RAID already exists, nothing to do", "raid_device", raidDevice)
+		return nil
+	}
+
+	for _, device := range devices {
+		deviceName := path.Base(device)
+		if _, err := os.Stat(device); os.IsNotExist(err) {
+			return fmt.Errorf("device %q does not exists", deviceName)
+		}
+
+		blockDevice, err := blkutils.IsBlockDevice(ctx, executor, device)
+		if err != nil {
+			return fmt.Errorf("can't check if device %q is block device: %w", deviceName, err)
+		}
+
+		if !blockDevice {
+			return fmt.Errorf("device %q is not a block device", deviceName)
+		}
+
+		if busy := isBusy(sysfsPath, device); busy {
+			return fmt.Errorf("device %q is busy", deviceName)
+		}
+
+		if systemPartition, err := isSystemPartition(ctx, executor, device); err != nil {
+			return fmt.Errorf("can't check if device %q is system partition: %w", deviceName, err)
+		} else if systemPartition {
```

Comment:
This is what current scylla_io_setup is checking. I rewrote all checks done there here. I don't know exact reason, but they were added probably because of one. I guess mdadm should complain when something is wrong, so I would be ok with removing them and relying on third party checks.

Reply:
> I don't know exact reason, but they were added probably because of one.

The person adding it there might have said the same thing :stuck_out_tongue_winking_eye:

Reply:
removed